test/puj/push/core_instructions_test.clj

```diff
@@ -34,10 +42,30 @@
       {:int '(50 3)}
       {:int '(150)}
 
+      :int-mod
+      {:int '(5 33 -12)}
+      {:int '(3 -12)}
+
+      :int-mod
+      {:int '(-5 33 -12)}
+      {:int '(-2 -12)}
+
+      :int-mod
+      {:int '(5 -33 -12)}
+      {:int '(2 -12)}
+
       :int-inc
       {:int '(50)}
       {:int '(51)}
 
+      :int-inc      ;; Test empty stack
+      {:int '()}
+      {:int '()}
```

Comment:
Not something that needs to change, but just an fyi.

Now that the code for popping arguments and pushing return values is centralized for all simple instructions, we don't need to test this edge case for every instruction. Instead we test it once here:

https://github.com/push-language/Puj/blob/daaa4adf5f6cd9535785bcc2ce970933ecd1cca0/test/puj/push/instruction_test.clj#L22

Reply:
Very cool! But, I can't figure out what that line does, without more context, and would have no idea that it tests (all? some? one?) instructions on an empty stack if I encountered it in the wild. @erp12, some of the testing code could use some more comments/docstrings please 😄

Reply:
Good point. I will work on that shortly in a separate PR.

The `instruction_test.clj` file doesn't test any of the instructions. Instead it tests the logic of the instruction abstractions (ie. `simple-instruction`). In the case of empty stacks (or insufficient args), we always expect the push-state to be unchanged and there is one code path that controls this behavior for all simple instructions. Thus, IMHO, we don't need to test any specific instructions.

This gets more complicated for more generic instruction types. For example the `state-to-state-instruction` has free range over the push-state and would call for more tests than a `simple-instruction`.

Reply:
Sounds good! Is it worth testing one instruction with empty stacks, to make sure that the actual execution of the instruction doesn't differ from this abstract testing without an instruction? If not, it would be fine to remove this!